### PR TITLE
Split interop client and server into components

### PR DESCRIPTION
**Pull request by yang-g**

### CODE REVIEW

**Comment by yang-g** [comment]
@fengli-google

File: build.json [hunk 1]
```diff
@@ -532,6 +532,36 @@
...
+    },
+    {
+      "name": "interop_server_lib",
+      "build": "private",
+      "language": "c++",
+      "src": [
+        "test/cpp/interop/server_helper.cc"
+      ],
+      "deps": [
+        "grpc++"
```
Comment (nicolasnoble) [inline build.json]: Why the asymmetry in the dependencies of both libs? This doesn't really affect anything except my ocd mind you.

Comment (yang-g) [inline build.json]: I changed it a bit to reflect what I use. Does it even matter? What is the general guideline to put things in deps of a library?

**Comment by yang-g** [comment]
I've updated build.json according to our offline discussion :)

**Pull request merged**